x-pack/plugins/apm/server/lib/errors/get_error_rate.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  ERROR_GROUP_ID,
+  PROCESSOR_EVENT,
+  SERVICE_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { getMetricsDateHistogramParams } from '../helpers/metrics';
+import { rangeFilter } from '../helpers/range_filter';
+import {
+  Setup,
+  SetupTimeRange,
+  SetupUIFilters,
+} from '../helpers/setup_request';
+
+export async function getErrorRate({
+  serviceName,
+  groupId,
+  setup,
+}: {
+  serviceName: string;
+  groupId?: string;
+  setup: Setup & SetupTimeRange & SetupUIFilters;
+}) {
+  const { start, end, uiFiltersES, client, indices } = setup;
+
+  const filter = [
+    { term: { [SERVICE_NAME]: serviceName } },
+    { range: rangeFilter(start, end) },
+    ...uiFiltersES,
+  ];
+
+  const aggs = {
+    response_times: {
+      date_histogram: getMetricsDateHistogramParams(start, end),
+    },
+  };
+
+  const getTransactionBucketAggregation = async () => {
+    const resp = await client.search({
+      index: indices['apm_oss.transactionIndices'],
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...filter,
+              { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
+            ],
+          },
+        },
+        aggs,
+      },
+    });
+    return {
+      totalHits: resp.hits.total.value,
+      responseTimeBuckets: resp.aggregations?.response_times.buckets,
+    };
+  };
+  const getErrorBucketAggregation = async () => {
+    const groupIdFilter = groupId
+      ? [{ term: { [ERROR_GROUP_ID]: groupId } }]
+      : [];
+    const resp = await client.search({
+      index: indices['apm_oss.errorIndices'],
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...filter,
+              ...groupIdFilter,
+              { term: { [PROCESSOR_EVENT]: ProcessorEvent.error } },
+            ],
+          },
+        },
+        aggs,
+      },
+    });
+    return resp.aggregations?.response_times.buckets;
+  };
+
+  const [transactions, errorResponseTimeBuckets] = await Promise.all([
+    getTransactionBucketAggregation(),
+    getErrorBucketAggregation(),
+  ]);
+
+  const transactionCountByTimestamp: Record<number, number> = {};
+  if (transactions?.responseTimeBuckets) {
+    transactions.responseTimeBuckets.forEach((bucket) => {
+      transactionCountByTimestamp[bucket.key] = bucket.doc_count;
+    });
+  }
+
+  const errorRates = errorResponseTimeBuckets?.map((bucket) => {
+    const { key, doc_count: errorCount } = bucket;
+    const transactionCount = transactionCountByTimestamp[key] || 1;
```

Comment:
why 1?

Reply:
Because I don't want to divide the errors by 0 or undefined.

Reply:
@cauemarcondes I'm glad you think about that so we don't get runtime exceptions.
However, I don't think setting the count to 1 is the way to go about it. If `transactionCount` is 0 or undefined `relativeRate` should be `undefined` (and possibly displayed as `N/A`)

Reply:
That sounds good to me. @formgeist ?

Reply:
@cauemarcondes Yes, that means we can't calculate the error rate IMO